org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/util/JavaProjectSetupUtil.java

```diff
@@ -81,32 +81,27 @@ public TextFile(String path, String content) {
 	 * creates a JarInputStream containing the passed text files. Each Pair<String
 	 */
 	public static InputStream jarInputStream(TextFile... files) {
-		try {
-			ByteArrayOutputStream out2 = new ByteArrayOutputStream();
-			JarOutputStream jo = new JarOutputStream(new BufferedOutputStream(out2));
+		try (ByteArrayOutputStream out2 = new ByteArrayOutputStream();
+				JarOutputStream jo = new JarOutputStream(new BufferedOutputStream(out2))) {
 			for (TextFile textFile : files) {
 				JarEntry je = new JarEntry(textFile.path);
 				jo.putNextEntry(je);
 				byte[] bytes = textFile.content.getBytes();
 				jo.write(bytes, 0, bytes.length);
 			}
-			jo.close();
 			return new ByteArrayInputStream(out2.toByteArray());
```

Comment:
This will probably miss bytes from the jo since it could still hold items in its buffer. Please add jo.flush(..) before reading the byte array

Reply:
Done.